GraphGym documentation (#112)
vrtex-team/pytorch_geometric: docs/source/notes/graphgym.rst

GraphGym - Manager for experiments
==================================

**Author: Jiaxuan You**

GraphGym is a platform for **designing and evaluating Graph Neural Networks (GNNs)**.
GraphGym is originally proposed in `Design Space for Graph Neural Networks, NeurIPS 2020 Spotlight <https://arxiv.org/abs/2011.08843>`__ and hosted in `Github repository <https://github.com/snap-stanford/GraphGym>`__.

> GraphGym has teamed up with PyTorch Geometric to provide an easy-to-use workflow for GNN development.

Reviewer: Shall we say that GraphGym API is likely going to change as we are continuously working on better and deeper integration with PyG? Similar to how PyTorch is doing it for [`torch.sparse`](https://pytorch.org/docs/stable/sparse.html).

> Highlights

Reviewer: You have some nice figures in the [original README](https://github.com/snap-stanford/GraphGym). It would be great to add them here as well.

----------

**1. Highly modularized pipeline for GNN**

- **Data:** Data loading, data splitting
- **Model:** Modularized GNN implementation
- **Tasks:** Node / edge / graph level GNN tasks
- **Evaluation:** Accuracy, ROC AUC, ...

**2. Reproducible experiment configuration**

- Each experiment is *fully described by a configuration file*

**3. Scalable experiment management**

- Easily launch *thousands of GNN experiments in parallel*
- *Auto-generate* experiment analyses and figures across random seeds and experiments.

**4. Flexible user customization**

- Easily *register your own modules*, such as data loaders, GNN layers, loss functions, etc.

Why GraphGym?
-------------

**TL;DR:** GraphGym is great for GNN beginners, domain experts and GNN researchers.

> **Scenario 1:** You are a beginner to GNN, who wants to understand how GNN works.

Reviewer: ```suggestion
**Scenario 1:** You are a beginner to GNNs, who wants to understand how GNNs works.
```

> You probably have read many exciting papers on GNN, and try to write your own GNN implementation.

Reviewer: ```suggestion
You probably have read many exciting papers on GNNs, and try to write your own GNN implementation.
```

> Using existing packages for GNN, you still have to code up the essential pipeline on your own.

Reviewer: ```suggestion
Even if using raw PyTorch Geometric, you still have to code up the essential pipeline on your own.
```
sounds a bit better :)

Author: Indeed! I mostly copy-paste from previous GraphGym docs
